packages/jest-leak-detector/src/index.ts

```diff
@@ -26,23 +28,33 @@ export default class LeakDetector {
       );
     }
 
-    let weak: typeof import('weak-napi');
+    if (globalThis.FinalizationRegistry) {
+      this._finalizationRegistry = new FinalizationRegistry(() => {
+        this._isReferenceBeingHeld = false;
+      });
 
-    try {
-      // eslint-disable-next-line import/no-extraneous-dependencies
-      weak = require('weak-napi');
-    } catch (err: any) {
-      if (!err || err.code !== 'MODULE_NOT_FOUND') {
-        throw err;
+      this._finalizationRegistry.register(value as object, undefined);
```

Comment:
Do we need to assign it to `this`? Can it just be function scoped instead?

Reply:
> Do we need to assign it to `this`? Can it just be function scoped instead?

When the instance of `FinalizationRegistry` is a local variable, it could be garbage collected, and the callback would be GCed and not be called too. So we need to ensure it has a referencer.

Reply:
> aha! interesting. can you add that as a code comment? 🙂

That's added!